unitTests/magicbox/SuggestionsManagerTest.ts

```diff
@@ -411,19 +415,61 @@ export function SuggestionsManagerTest() {
             return Utils.resolveAfter(previewsPromisesWaitTimes[suggestionId], previewsBySuggestion[suggestionId]);
           }
 
+          const delayBeforePopulatePreviews = 150;
+          function bindUpdateResultPreviewsManagerOptionsEvent() {
+            $$(env.root).on(
+              ResultPreviewsManagerEvents.updateResultPreviewsManagerOptions,
+              (_, args: IUpdateResultPreviewsManagerOptionsEventArgs) => {
```

Comment:
I would simplify the mock here. Right now we are re-implementing the logic. Instead, we only need a spy on the method in charge of handling the `ResultPreviewsManagerEvents.updateResultPreviewsManagerOptions`, that returns a value we configure. e.g.

```
function setReviewPopulationDelay(delay: number) {
    spyOn(classInstance, 'handlerMethod').and.returnValue(delay);
}

Reply:
Good point, I'll replace it with a spy.